mcstatus/status_response.py

```diff
@@ -0,0 +1,326 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Union
+
+STYLE_MAP = {
+    "bold": "l",
+    "italic": "o",
+    "underlined": "n",
+    "obfuscated": "k",
+    "color": {
+        "dark_red": "4",
+        "red": "c",
+        "gold": "6",
+        "yellow": "e",
+        "dark_green": "2",
+        "green": "a",
+        "aqua": "b",
+        "dark_aqua": "3",
+        "dark_blue": "1",
+        "blue": "9",
+        "light_purple": "d",
+        "dark_purple": "5",
+        "white": "f",
+        "gray": "7",
+        "dark_gray": "8",
+        "black": "0",
+    },
+}
+
+
+@dataclass
+class AbstractDataclass(ABC):
+    """Abstract base class for all dataclasses.
+
+    We can't use just `ABC` because we need raise TypeError if we try to
+    instantiate an abstract dataclass. Class which inherit from `ABC` raises
+    it only if it has abstract methods.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if cls == AbstractDataclass or cls.__bases__[0] == AbstractDataclass:
+            raise TypeError("Cannot instantiate abstract class.")
+        return super().__new__(cls)
+
+
+@dataclass
+class MCServerResponse(AbstractDataclass):
+    """Dataclass for storing cross-platform status answer object."""
```

Comment:
This isn't a class for cross-platform result, since it doesn't combine the java and bedrock results into one, it's merely a class for shared values between both platforms, so the docstring shouldn't describe it as cross-platform response. 

It also isn't for storing a "status answer object", instances of this class are status answer objects, it isn't a wrapper around the old response objects (anymore).

We also don't need to inform the library users that this was a dataclass, that's not relevant and doesn't need to be in the docstring.
```suggestion
    """Class for storing shared data from a status response."""
```

Many other classes have some of the same issues in their docstrings, so this applies for all of them.

Reply:
Fixed in fa7beed82688eab84709c9928a69ff7bffe58431.